dds/idl/be_global.cpp

```diff
@@ -602,3 +613,147 @@ BE_GlobalData::get_include_block(BE_GlobalData::stream_enum_t which)
 
   return ret.str();
 }
+
+void
+BE_GlobalData::cache_annotations()
+{
+  if (idl_global->idl_version_ >= IDL_VERSION_4) {
+    UTL_Scope* root = idl_global->scopes().bottom();
+    topic_annotation_ =
+      AST_Annotation_Decl::narrow_from_decl(
+        root->lookup_by_name("::@topic"));
+    key_annotation_ = AST_Annotation_Decl::narrow_from_decl(
+        root->lookup_by_name("::@key"));
+    nested_annotation_ = AST_Annotation_Decl::narrow_from_decl(
+        root->lookup_by_name("::@nested"));
+    default_nested_annotation_ = AST_Annotation_Decl::narrow_from_decl(
+        root->lookup_by_name("::@default_nested"));
+ }
+}
+
+/**
+ * @brief determines if we should treat a given node as a topic type. This could
+ * mean that it is explicitly a topic type or that @default_nested(FALSE) or
+ * --default_nested = false
+ * @param node the incomming node
+ * @return if type support should be generated
+ * @see BE_GlobalData::is_topic_type(AST_Decl* node)
+ * @see BE_GlobalData::is_default_nested(AST_Decl* node)
+ * @author ceneblock
+ */
+bool BE_GlobalData::treat_as_topic(AST_Decl *node)
+{
+  return is_topic_type(node) || !is_nested_type(node);
+}
+
+/**
+ * @brief determins if a given node is nested by default or not. This may be set
+ * by the module level of @default_nesed or through the --default_nested=
+ * @param node the node to be inspected
+ * @return if the node is default nested or not
+ * @note false means it should be considered nested.
+ * @author ceneblock
+ */
+bool
+BE_GlobalData::is_default_nested(AST_Decl* node)
+{
+  if(node->defined_in()){
```

Comment:
space after keyword (if)

Reply:
I won't harp on these anymore, but there are several more, so maybe do a global search / replace in touched files?

Reply:
I will be fixing the style.